addons/mrp_account/models/product.py

```diff
@@ -50,7 +50,7 @@ def _compute_average_price(self, qty_invoiced, qty_to_invoice, stock_moves):
         dummy, bom_lines = bom.explode(self, 1)
         bom_lines = {line: data for line, data in bom_lines}
         value = 0
-        for move in stock_moves:
+        for move in stock_moves.filtered(lambda m: not (m.origin_returned_move_id and sum(m.stock_valuation_layer_ids.mapped('quantity')) >= 0)):
```

Comment:
The correct fix is probably to do a group by bom_line id first and call _compute_average_price with the correct set of moves